### PR TITLE
Update the launch scripts for lifecycle nodes

### PR DESCRIPTION
**Pull request by mjeronimo**
## Description
* Use ExecuteProcess directly for the lifecycle nodes to avoid the broken LifecycleNode class in the Python launch. The Node and LifecycleNode Python classes should not assume that there is one node per process. They should also not rename other nodes created in the process. 
* Also, specify use_sim_time in the parameters file to avoid having to set it manually

## Future work
* Could fix the Node and LifecycleNode Python classes in the launch package
* Could separate out the use_sim_time parameter settings into a separate parameters file and use cascading parameters file to specify whether or not to use_sim_time. In other words, there is a base parameters file that is used and can be specified on the command line. Then, based on a command-line argument, use_sim_time, could use a second parameters file; either one where all the nodes have use_sim_time=true or another one where use_sim_time=false.

### CODE REVIEW

File: nav2_bringup/launch/test_map.yaml [hunk 1]
```diff
@@ -1,8 +1,6 @@
-map_type: "occupancy"
-image: "test_map.pgm"
+image: test_map.pgm
 resolution: 0.050000
-origin: [-15.400000, -12.200000, 0.000000]
-free_thresh: 0.196
-occupied_thresh: 0.65
+origin: [-10.000000, -10.000000, 0.000000]
```
Comment (crdelsey) [inline nav2_bringup/launch/test_map.yaml]: Why did you change the test map?

File: nav2_bringup/launch/nav2_params.yaml [hunk 2]
```diff
@@ -31,28 +82,81 @@ DwbController:
...
-        enabled: False
-      always_send_full_costmap: True
-      observation_sources: scan
-      scan:
-        topic: /scan
-        max_obstacle_height: 2.0
-        clearing: True
-        marking: True
+
+dwb_controller_local_costmap:
```
Comment (crdelsey) [inline nav2_bringup/launch/nav2_params.yaml]: Why the node name change? I also don't see anywhere in the launch file where you made an actual change to the node name.

We also want to keep the costmaps in separate namespaces otherwise their topics will interfere with each other.

File: nav2_bringup/launch/nav2_params.yaml [hunk 3]
```diff
@@ -1,8 +1,59 @@
-map_server:
+amcl:
   ros__parameters:
-    yaml_filename: "test_map.yaml"
-DwbController:
+    use_sim_time: True
```
Comment (crdelsey) [inline nav2_bringup/launch/nav2_params.yaml]: I thought you were going to put the `use_sim_time` parameters in a seperate YAML file so we could choose whether to include it or not?

File: nav2_bringup/launch/nav2_bringup_launch.py [hunk 4]
```diff
@@ -1,64 +1,160 @@
...
+
+    # Specify the actions
+
+    start_gazebo_cmd = launch.actions.ExecuteProcess(
+        condition=IfCondition(use_simulation),
+        cmd=[simulator, '-s', 'libgazebo_ros_init.so', world, ['__params:=', params_file]],
+        cwd=[launch_dir], output='screen')
+
+    start_robot_state_publisher_cmd = launch.actions.ExecuteProcess(
+        condition=IfCondition(use_simulation),
```
Comment (crdelsey) [inline nav2_bringup/launch/nav2_bringup_launch.py]: We're going to want the robot state publisher if we're running on the real robot as well. You can probably get rid of the if use_simulation check for this one.

File: nav2_bringup/launch/bt_navigator.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<root main_tree_to_execute="MainTree">
```
Comment (crdelsey) [inline nav2_bringup/launch/bt_navigator.xml]: Is this new BT tree file related to the launch script change or was this not supposed to be in this PR?

File: nav2_bringup/launch/nav2_params.yaml (hunk 3 above)
Comment (mjeronimo) [reply nav2_bringup/launch/nav2_params.yaml]: Wasn't planning to do that as part of this PR. We can do that. I'll add it to the list of TODO items for the integration branch once we get these changes in.

File: nav2_bringup/launch/bt_navigator.xml (hunk 5 above)
Comment (mjeronimo) [reply nav2_bringup/launch/bt_navigator.xml]: I added it so that the launch directory had everything needed to launch the stack - the launch script, params file, the map yaml and pgm, and the BT navigator XML file.

File: nav2_bringup/launch/nav2_bringup_launch.py (hunk 4 above)
Comment (mjeronimo) [reply nav2_bringup/launch/nav2_bringup_launch.py]: Ok, fixed.

File: nav2_bringup/launch/nav2_params.yaml (hunk 2 above)
Comment (mjeronimo) [reply nav2_bringup/launch/nav2_params.yaml]: To sort out for my self the various node names and parameters for myself, I used a <node_name>_<composed_node_name> scheme. I can change the name back. 

Why would I have to change the node name in the launch file? As long as the name in the params yaml file matches the node name, you're all set. 

Good point about the namespaces. Let's talk so I can make sure to update this correctly.

File: nav2_bringup/launch/test_map.yaml (hunk 1 above)
Comment (mjeronimo) [reply nav2_bringup/launch/test_map.yaml]: I wasn't sure about the interaction of the map and the world. I had gazebo coming up with the other map, so I changed at the time to sync them up. Let me get your help to fix this correctly.

Comment (mjeronimo) [reply nav2_bringup/launch/test_map.yaml]: We discussed and agreed to go with the turtlebot map.

File: nav2_bringup/launch/nav2_params.yaml (hunk 2 above)
Comment (mjeronimo) [reply nav2_bringup/launch/nav2_params.yaml]: I'd like to fix the node names in a separate PR since I'll also have to update nav2_dwb_controller and nav2_navfn_planner in concert with this.

**Comment by mjeronimo** [comment]
@crdelsey Yes, that's my understanding. Let's integrate so that we can get people going using this branch and then we'll address the shortcomings.
